### ISSUE TITLE
W1b: `Boxes.compute_area` returns `(w - 1) * (h - 1)` — 0.0 for a valid 2×1 box, and `filter_boxes_by_area` then zeroes it

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4010]
`Boxes.compute_area` applies the shoelace formula to the stored **inclusive** vertices. A box of width `w` and height `h` (in the class's own `get_boxes_shape` terms) has vertices spanning `w - 1` and `h - 1`, so the polygon area is `(w - 1) * (h - 1)`. A valid 2×1 box has area 0.0.

## Repro

```python
import torch
from kornia.geometry.boxes import Boxes

b = Boxes.from_tensor(torch.tensor([[[1., 1., 3., 2.]]]), mode="xyxy")     # w=2, h=1
print("get_boxes_shape:", [t.tolist() for t in b.get_boxes_shape()])
print("compute_area:", b.compute_area().tolist())
print("filter_boxes_by_area(min_area=1):", b.filter_boxes_by_area(min_area=1.0).data.tolist())

doc = Boxes(torch.tensor([[[1., 1.], [4., 1.], [4., 3.], [1., 3.]]]))          # the to_mask docstring example
print("doc example get_boxes_shape:", [t.tolist() for t in doc.get_boxes_shape()], " compute_area:", doc.compute_area().tolist())
```

## Observed

```
get_boxes_shape: [[[1.0]], [[2.0]]]
compute_area: [[0.0]]
filter_boxes_by_area(min_area=1): [[[[0.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.0]]]]     <- a valid box, deleted
doc example get_boxes_shape: [[3.0], [4.0]]  compute_area: [6.0]                             <- 3 x 4 = 12 expected
```

`tests/geometry/test_boxes.py::test_compute_area` never caught this because it constructs raw quadrilaterals and checks them against the shoelace value — it never asks whether a `from_tensor` box's area agrees with its own `get_boxes_shape`.

## Expected

`compute_area()` equals `heights * widths` from `get_boxes_shape()` for axis-aligned boxes (and the shoelace area of the *inclusive* polygon — each vertex extended by one — for general quadrilaterals). Every caller's numbers change, so the fix is a semantic change even though the current value is indefensible for a 2×1 box.

## Classification

**Broken output by the class's own contract** (`get_boxes_shape` says 1×2, `compute_area` says 0) with a side effect in `filter_boxes_by_area`. Root cause is the inclusive representation, i.e. the same root as #3934, so the repair belongs in the same window unless the maintainer rules the 0-area case a plain bug worth fixing early. Idiom 3 (the computed quantity does not match the documented contract).

## Disposition

n/a — single symbol. Cross-links: #3934 (same root cause), the sibling issue on the exclusive export (#4009).

## Environment

kornia 0.9.0rc1 @ `0685b1f5` (main), torch 2.9.1, python 3.11, cpu. Found during the bbox/boxes conventions audit (batch 4, tracking issue #4020); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
